### PR TITLE
HB-05

### PR DESCRIPTION
**Pull request by pavlokhomiak**

### CODE REVIEW

**Review by ilya449 — APPROVED** [review]

**Comment by ilya449** [comment]
Good job!

File: src/main/java/com/dev/cinema/util/HashUtil.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(SHA_ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new DataProcessingException("No such algorithm", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/util/HashUtil.java]:
```suggestion
            throw new RuntimeException("No such algorithm", e);
```

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Inject
+    UserDao userDao;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: what about access modifier?

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password)
+            throws AuthenticationException {
+        Optional<User> userFromDB = userService.findByEmail(email);
+        if (userFromDB.isPresent()) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: let's simplify it and make check with only one if clause

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**